lib/active_record/belongs_to_if/preloader_extension.rb

```diff
@@ -26,6 +26,32 @@ def preload(*)
 
       private
 
+      # @note Override to change `records_by_owner` so that empty array is used.
+      def associated_records_by_owner(preloader)
+        owners_map = owners_by_key
+        owner_keys = owners_map.keys.compact
+
+        # Each record may have multiple owners, and vice-versa
+        records_by_owner = Hash.new do |hash, key|
+          hash[key] = []
+        end
```

Comment:
In ActiveRecord 4.2.0, this Hash was changed with nil-based unexpectedly for me.